delivery_local_pickup/models/delivery_carrier.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, fields, models
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    delivery_type = fields.Selection(selection_add=[("local_pickup", "Local pickup")])
+    local_address_id = fields.Many2one(
+        comodel_name="res.partner", string="Local address"
+    )
+
+    def local_pickup_send_shipping(self, pickings):
+        return [self.local_pickup_create_shipping(p) for p in pickings]
+
+    def local_pickup_create_shipping(self, picking):
+        return {"exact_price": False, "tracking_number": _("Local pickup")}
+
+    def local_pickup_get_tracking_link(self, picking):
+        config_parameter = self.env["ir.config_parameter"].sudo()
+        base_url = config_parameter.get_param(
+            "web.base.url", default="http://localhost:8069"
```

Comment:
This is not necessarily true if the website doesn't have the same url, like for example multi company/multi website. Maybe a proper way would be to add a website_id field or at least default to the proper company website url. This could be added to the ROADMAP for now anyway

Reply:
I have changed this a bit to use the carrier's website if it is defined.

Reply:
I don't think you should build the URL this way, but there should be a method that builds it. Take into account that the `website_id` field is only available when you install `website_delivery`, so we can't depend on it.

Reply:
Ok, I had added the `website_delivery` dependency to use the `website_id` field and also for this https://github.com/OCA/delivery-carrier/pull/415#discussion_r710059907 but if you want better we can remove it.

Ok, I will try to find some odoo core method to build the URL although if you know any it would be great.

Reply:
Please don't. If any, we will need to build a `website_delivery_local_pickup` glue module for that, that returns a different URL if a website is put. I don't know which is the method for the base to return the current URL, but `mail` uses for building the URL for the buttons `"View <object>"` that we see on the generated mails, so please search a bit.

Reply:
Odoo does it this way: https://github.com/odoo/odoo/blob/13.0/addons/mail/models/mail_thread.py#L455
It is already changed to use something similar.

Reply:
Yes, that's why I said it could be put on the roadmap. Anyway, we could check that the attribute exists and behave accordinly if so to avoid the hard dependency or a glue module. `website_sale_delivery` is auto installed with this module's dependencies after all: https://github.com/odoo/odoo/blob/13.0/addons/website_sale_delivery/__manifest__.py#L22

Reply:
I think this function would be enough.

In the future, another addon `website_delivery_local_pickup` (delivery_local_pickup + website_sale_delivery) would implement the function to return the URL according to the `website_id` field and would add the `website_description` field of the carrier to the website.

Reply:
Well, I like the idea of checking `if "website_id" in self._fields` for not having to have a glue module.